tests/unit/test_spark_utils.py

```diff
@@ -2,11 +2,60 @@
 
 import pytest
 
-from linker.utilities.spark_utils import find_spark_master_url
+from linker.utilities.spark_utils import (
+    build_cluster_launch_script,
+    find_spark_master_url,
+)
 
 
-def test_logfile_reader_fails_after_10_attempts():
-    """Test that there is not infinite loop when trying to read the logfile"""
+def test_build_cluster_launch_script(test_dir):
+    results_dir = Path("/some/results/dir")
+    diagnostics_dir = Path(test_dir)  # shell script is saved to diagnostics dir
+    input_data = [Path("/some/input/data"), Path("some/other/input/data")]
+    launcher = build_cluster_launch_script(
+        results_dir=results_dir,
+        diagnostics_dir=diagnostics_dir,
+        input_data=input_data,
+    )
+    assert launcher
+    assert launcher.name.startswith(f"{test_dir}/spark_cluster_launcher_")
+    assert launcher.name.split(".")[-1] == "sh"
+
+    # Don't check all of the contents but check a few key lines exist
+    with open(Path(test_dir) / launcher.name, "r") as f:
+        lines = f.readlines()
+        assert lines[1] == "#!/bin/bash\n"
+        assert "SPARK_MASTER_PORT=28508\n" in lines
+        assert "SPARK_MASTER_WEBUI_PORT=28509\n" in lines
```

Comment:
nit: perhaps add one for 
`SINGULARITY_IMG={CONTAINER_DIR}/spark_cluster.sif`?
This is the only other thing that is not necessarily a static part of the script.